src/fty_mdns_sd_mapping.cc

```diff
@@ -0,0 +1,272 @@
+/*  =========================================================================
+    fty_mdns_sd_mapping - 42ity mdns sd mapping class
+
+    Copyright (C) 2019 Eaton
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+    =========================================================================
+*/
+
+#include "fty_mdns_sd_classes.h"
+
+#include <cxxtools/jsonserializer.h>
+#include <cxxtools/jsondeserializer.h>
+
+//  --------------------------------------------------------------------------
+// ServiceDeviceMapping
+//
+
+ServiceDeviceMapping::ServiceDeviceMapping():
+    m_serviceHostname(),
+    m_serviceName(),
+    m_serviceAddress(),
+    m_servicePort(),
+    m_extendedInfo()
+{}
+
+ServiceDeviceMapping::ServiceDeviceMapping(std::string hostname, std::string name, std::string address, std::string port, ExtendedInfoMapping extendedInfo) :
+    m_serviceHostname(hostname),
+    m_serviceName(name),
+    m_serviceAddress(address),
+    m_servicePort(port),
+    m_extendedInfo(extendedInfo)
+{}
+
+void ServiceDeviceMapping::fillSerializationInfo(cxxtools::SerializationInfo& si) const
+{
+    try {
+        si.addMember(SERVICE_HOST_NAME_ENTRY) <<= m_serviceHostname;
+        si.addMember(SERVICE_NAME_ENTRY) <<= m_serviceName;
+        si.addMember(SERVICE_ADDRESS_ENTRY) <<= m_serviceAddress;
+        si.addMember(SERVICE_PORT_ENTRY) <<= m_servicePort;
+        si.addMember(SERVICE_EXTENDED_INFO_ENTRY).setCategory(cxxtools::SerializationInfo::Object);
+        cxxtools::SerializationInfo *extended = si.findMember(SERVICE_EXTENDED_INFO_ENTRY);
+        if (extended) {
+            for (auto &info : m_extendedInfo) {
+                extended->addMember(info.first) <<= info.second;
+            }
+        }
```

Comment:
Calling `findMember()` and checking for existence of node is unnecessary, consider something like this instead:
```c++
const auto& extendedSi = si.addMember(SERVICE_EXTENDED_INFO_ENTRY);
extendedSi.setCategory(cxxtools::SerializationInfo::Object);
for (const auto& info : m_extendedInfo) {
    extendedSi.addMember(info.first) <<= info.second;
}
```